### ISSUE TITLE
F* doesn't seem to call Z3 properly

### ISSUE THREAD

**Issue by msprotz** [FStarLang/FStar#352]
While testing the new OPAM packaging:

```bash
jonathan@ramona:/tmp/f $ bin/fstar.exe share/fstar/examples/hello/hello.fst 
/bin/sh: 4: SMT.RELEVANCY=2: not found
Verifying module: Hello
All verification conditions discharged successfully
```

- F* doesn't call z3 properly
- F* seems to consider this a success (even though I believe z3 never checked anything)

```
jonathan@ramona:/tmp/f $ which z3
/usr/bin/z3
```

**Comment by catalin-hritcu** [comment]
seems like a broken F* binary

**Comment by catalin-hritcu** [comment]
the cause can be this: an OCaml binary can't currently produce a correct OCaml binary (#335)

**Comment by catalin-hritcu** [comment]
This should be fixed now that #335 is fixed.
